DOC-2268: Add new documentation for `ExportWord` Plugin. (#3072)
tinymce/tinymce-docs: modules/ROOT/pages/exportword.adoc

= Export to Word
:navtitle: exportword
:description: The export to Word feature lets you generate a .docx file directly from the editor.
:description_short: Generate a .docx file directly from the editor.
:keywords: plugin, exportword.
:pluginname: Export to Word
:plugincode: exportword
:plugincategory: premium

include::partial$misc/admon-paid-addon-pricing.adoc[]

include::partial$misc/admon-requires-7.0v.adoc[]

The export to Microsoft Word feature collects the HTML generated with the `tinymce.editor.getContent()` method and combines it with the default editor content styles along with the styles provided in the configuration. The combined content and styles are then transmitted to the {productname} Cloud Services HTML to DOCX converter service. Following this, the service generates a Word file, which is subsequently returned to the user’s browser, enabling them to save it in the Word format onto their disk.

== Interactive example

[NOTE]
.{pluginname} demo
====

> Demos containing the {pluginname} plugin are currently only available on our main website. Check out https://www.tiny.cloud/tinymce/features/exportword/[the {pluginname} demo], or the https://www.tiny.cloud/[demo on our home page], to try it out today.

Author: ```suggestion
Demos containing the {pluginname} plugin are currently only available on our main website. Check out https://www.tiny.cloud/tinymce/features/export-word/[the {pluginname} demo], or the https://www.tiny.cloud/[demo on our home page], to try it out today.
```

====

> [NOTE]
> The generated `.docx` file may not be fully compatible with older versions of Word.
> 

Author: ```suggestion
[NOTE]
The generated `.docx` file may not be fully compatible with older versions of Word.

[NOTE]
xref:introduction-to-mediaembed.adoc[Enhanced Media Embed] will not be included in the exported document.
```

:plugincode: exportword
:plugincategory: premium

include::partial$misc/admon-paid-addon-pricing.adoc[]

include::partial$misc/admon-requires-7.0v.adoc[]

The export to Microsoft Word feature collects the HTML generated with the `tinymce.editor.getContent()` method and combines it with the default editor content styles along with the styles provided in the configuration. The combined content and styles are then transmitted to the {productname} Cloud Services HTML to DOCX converter service. Following this, the service generates a Word file, which is subsequently returned to the user’s browser, enabling them to save it in the Word format onto their disk.

== Interactive example

[NOTE]
.{pluginname} demo
====
Demos containing the {pluginname} plugin are currently only available on our main website. Check out https://www.tiny.cloud/tinymce/features/export-word/[the {pluginname} demo], or the https://www.tiny.cloud/[demo on our home page], to try it out today.
====

[NOTE]
The generated `.docx` file may not be fully compatible with older versions of Word.

[NOTE]
xref:introduction-to-mediaembed.adoc[Enhanced Media Embed] will not be included in the exported document.

> == Basic setup

Author: ```suggestion

== Basic setup
```

= Export to Word
:navtitle: exportword
:description: The export to Word feature lets you generate a .docx file directly from the editor.
:description_short: Generate a .docx file directly from the editor.
:keywords: plugin, exportword.
:pluginname: Export to Word
:plugincode: exportword
:plugincategory: premium

> include::partial$misc/admon-paid-addon-pricing.adoc[]

Author: ```suggestion
include::partial$misc/admon-export-word-paid-addon-pricing.adoc[]
```

== Interactive example

[NOTE]
.{pluginname} demo
====
Demos containing the {pluginname} plugin are currently only available on our main website. Check out https://www.tiny.cloud/tinymce/features/export-word/[the {pluginname} demo], or the https://www.tiny.cloud/[demo on our home page], to try it out today.
====

[NOTE]
.{pluginname} known issues
====
* The generated `.docx` file may not be fully compatible with older versions of Word.
* xref:introduction-to-mediaembed.adoc[Enhanced Media Embed] will not be included in the exported document.
====

== Basic setup

To add the {pluginname} plugin to the editor, add `{plugincode}` to the `plugins` option in the editor configuration.

For example:

[source,js]
----
tinymce.init({
  selector: 'textarea',
  plugins: 'exportword',
  toolbar: 'exportword',
  exportword_service_url: '<service_URL>'
});
----

== Options

The following configuration options affect the behavior of the {pluginname} plugin.

> include::partial$configuration/{plugincode}.adoc[][leveloffset=+1]

Reviewer: ```suggestion
include::partial$configuration/{plugincode}.adoc[leveloffset=+1]
```
fix nav bar levels

= Export to Word
:navtitle: exportword
:description: The export to Word feature lets you generate a .docx file directly from the editor.
:description_short: Generate a .docx file directly from the editor.
:keywords: plugin, exportword.
:pluginname: Export to Word

> // values for <tier>
> // opensource plugin = tieroneplan
> // premium plugin    = tiertwoplan or tierthreeplan or enterpriseplan

Reviewer 2: Just leaving a comment so we don't forget. I think it can be removed?
```suggestion
```

:pluginminimumplan: <tier>
:plugincode: exportword
:plugincategory: premium

include::partial$misc/admon-paid-addon-pricing.adoc[]

include::partial$misc/admon-requires-7.0v.adoc[]

> The export to Microsoft Word feature collects the HTML generated with the `tinymce.editor.getContent()` method and the default editor content styles combined with the styles provided by you in the configuration. It then sends them to the {productname} Cloud Services HTML to DOCX converter service. The service generates a Word file and returns it to the user’s browser so they can save it in the Word format on their disk.

Reviewer 3: Do we usually write docs in second person form and use `you` or should this be reformulated to be `styles provided by the integrator` or `custom defined styles` or something?

Author: ```suggestion
The export to Microsoft Word feature collects the HTML generated with the `tinymce.editor.getContent()` method and combines it with the default editor content styles along with the styles provided in the configuration. The combined content and styles are then transmitted to the {productname} Cloud Services HTML to DOCX converter service. Following this, the service generates a Word file, which is subsequently returned to the user’s browser, enabling them to save it in the Word format onto their disk.
```

= Export Word
:navtitle: exportword
:description: The export to Word feature lets you generate a .docx file directly from the editor.
:description_short: Generate a .docx file directly from the editor.
:keywords: plugin, exportword.
:pluginname: Export to Word
// values for <tier>
// opensource plugin = tieroneplan
// premium plugin    = tiertwoplan or tierthreeplan or enterpriseplan
:pluginminimumplan: <tier>
:plugincode: exportword
:plugincategory: premium

include::partial$misc/admon-paid-addon-pricing.adoc[]

include::partial$misc/admon-requires-7.0v.adoc[]

The export to Microsoft Word feature collects the HTML generated with the `tinymce.editor.getContent()` method and the default editor content styles combined with the styles provided by you in the configuration. It then sends them to the {productname} Cloud Services HTML to DOCX converter service. The service generates a Word file and returns it to the user’s browser so they can save it in the Word format on their disk.

[NOTE]
The generated `.docx` file may not be fully compatible with older versions of Word.

> The complementary pagination feature allows you to see where page breaks would be after you export the document to Word. Thanks to the live preview, the user can fine-tune the structure of the output document when editing it. The pagination feature also shows you the page count and lets you navigate between the document pages.

Reviewer 4: I don't know about any live preview for this feature. It seems this is a left over and should be removed ?

Author: ```suggestion
```

== Interactive example

liveDemo::{plugincode}[]

== Basic setup

To add the {pluginname} plugin to the editor, add `{plugincode}` to the `plugins` option in the editor configuration.

For example:

[source,js]
----
tinymce.init({
  selector: 'textarea',
  plugins: 'exportword',

>   toolbar: 'exportword',

Reviewer 4: We need to add the required `exportword_service_url` option here

});
----

== Options

The following configuration options affect the behavior of the {pluginname} plugin.

include::partial$configuration/{plugincode}.adoc[][leveloffset=+1]

== Commands

The {pluginname} plugin provides the following {productname} commands.

include::partial$commands/{plugincode}-cmds.adoc[][leveloffset=+1]

== Events

The {pluginname} plugin provides the following events.

> include::partial$events/{plugincode}-events.adoc[][leveloffset=+1]

Reviewer 4: We don't have any events in this plugin so this section should be removed in my opinion

Author: ```suggestion
```

= Export Word
:navtitle: exportword
:description: The export to Word feature lets you generate a .docx file directly from the editor.
:description_short: Generate a .docx file directly from the editor.
:keywords: plugin, exportword.
:pluginname: Export to Word
// values for <tier>
// opensource plugin = tieroneplan
// premium plugin    = tiertwoplan or tierthreeplan or enterpriseplan
:pluginminimumplan: <tier>
:plugincode: exportword
:plugincategory: premium

include::partial$misc/admon-premium-plugin.adoc[]

> 

Author: - [ ] Update placeholder for self-hosted users when confirmed.
```suggestion

[NOTE]
This feature is currently **only available** for sales managed `self-hosted` customers. A cloud-based option for self-served customers will be available in 2024.
```

////
include::partial$misc/admon-requires-7.0v.adoc[]
////

The export to Microsoft Word feature collects the HTML generated with the `tinymce.editor.getContent()` method and the default editor content styles combined with the styles provided by you in the configuration. It then sends them to the {productname} Cloud Services HTML to DOCX converter service. The service generates a Word file and returns it to the user’s browser so they can save it in the Word format on their disk.

[NOTE]
The generated `.docx` file may not be fully compatible with older versions of Word.

The complementary pagination feature allows you to see where page breaks would be after you export the document to Word. Thanks to the live preview, the user can fine-tune the structure of the output document when editing it. The pagination feature also shows you the page count and lets you navigate between the document pages.

== Interactive example

> liveDemo::{plugincode}[]

Author: @hamza0867 Can you assist with updating these demos when after this is merged and `7-dev` is up and running.

Author: Removed, and will replace this with link to website.

== Basic setup

To add the {pluginname} plugin to the editor, add `{plugincode}` to the `plugins` option in the editor configuration.

For example:

[source,js]
----
tinymce.init({
  selector: 'textarea',
  plugins: 'code exportword',
  toolbar: 'exportword code',
  content_css: '../css/demo.css'
});
----

== Options

The following configuration options affect the behavior of the {pluginname} plugin.

include::partial$configuration/{plugincode}.adoc[][leveloffset=+1]

== Commands

The {pluginname} plugin provides the following {productname} commands.

include::partial$commands/{plugincode}-cmds.adoc[][leveloffset=+1]

// == Events

// The {{pluginname}} plugin provides the following events.

// include::partial$events/{plugincode}-events.adoc[]

// == APIs

// The {{pluginname}} plugin provides the following APIs.

> // include::partial$plugin-apis/{plugincode}-apis.adoc[]

Author: @hamza0867 Do we need to add anything here at this stage?
